mesonbuild/cmake/interpreter.py

```diff
@@ -784,7 +784,6 @@ def __init__(self, build: 'Build', subdir: Path, src_dir: Path, install_prefix:
         self.languages         = []  # type: T.List[str]
         self.targets           = []  # type: T.List[ConverterTarget]
         self.custom_targets    = []  # type: T.List[ConverterCustomTarget]
-        self.trace             = CMakeTraceParser('', Path('.'), self.env)  # Will be replaced in analyse
```

Comment:
Maybe `self.trace: T.Optional[CMakeTraceParser] = None` instead? I am not a huge fan of introducing `self.X` variables in methods other than `__init__` (though this would not be a deal breaker if my suggestion introduces other problems).

Reply:
Try latest change on for size.